xtable-spark-plugin/src/main/java/org/apache/xtable/spark/XTableSparkCatalog.java

```diff
@@ -0,0 +1,506 @@
+import org.apache.spark.sql.catalyst.analysis.NoSuchNamespaceException;
+import org.apache.spark.sql.catalyst.analysis.NoSuchTableException;
+import org.apache.spark.sql.catalyst.analysis.NonEmptyNamespaceException;
+import org.apache.spark.sql.catalyst.analysis.TableAlreadyExistsException;
+import org.apache.spark.sql.connector.catalog.CatalogPlugin;
+import org.apache.spark.sql.connector.catalog.Identifier;
+import org.apache.spark.sql.connector.catalog.NamespaceChange;
+import org.apache.spark.sql.connector.catalog.SupportsNamespaces;
+import org.apache.spark.sql.connector.catalog.Table;
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.apache.spark.sql.connector.catalog.TableChange;
+import org.apache.spark.sql.connector.expressions.Transform;
+import org.apache.spark.sql.hudi.catalog.HoodieCatalog;
+import org.apache.spark.sql.types.StructType;
+import org.apache.spark.sql.util.CaseInsensitiveStringMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.iceberg.aws.glue.GlueCatalog;
+import org.apache.iceberg.spark.SparkCatalog;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import org.apache.xtable.catalog.TableFormatUtils;
+import org.apache.xtable.model.storage.TableFormat;
+
+import software.amazon.awssdk.services.glue.GlueClient;
+import software.amazon.awssdk.services.glue.model.GetTableRequest;
+import software.amazon.awssdk.services.glue.model.GetTableResponse;
+import software.amazon.awssdk.services.glue.model.GlueException;
+import software.amazon.awssdk.services.glue.model.StorageDescriptor;
+
+/**
+ * XTable Spark Catalog Plugin - Unified catalog implementation for both Hudi and Iceberg tables.
+ * This plugin leverages XTable's existing format detection capabilities and provides a seamless
+ * experience for querying mixed table formats through a single Spark catalog.
+ */
+public class XTableSparkCatalog implements CatalogPlugin, TableCatalog, SupportsNamespaces {
+
+  private static final Logger LOG = LoggerFactory.getLogger(XTableSparkCatalog.class);
+
+  private String catalogName;
+  private SparkCatalog icebergCatalog;
+  private HoodieCatalog hudiCatalog;
+  private GlueClient glueClient;
+  private CaseInsensitiveStringMap options;
+
+  @Override
+  public void initialize(String name, CaseInsensitiveStringMap options) {
+    this.catalogName = name;
+    this.options = options;
+
+    LOG.info("Initializing XTable Spark Catalog Plugin with name: {}", name);
+
+    this.glueClient = buildGlueClient(options);
+
+    SparkSession spark = SparkSession.active();
+    spark.conf().set("hoodie.schema.on.read.enable", "true");
+
+    initializeIcebergCatalog(name, options);
+    initializeHudiCatalog(options);
+  }
+
+  private GlueClient buildGlueClient(CaseInsensitiveStringMap options) {
+    String region =
+        options.getOrDefault("glue.region", options.getOrDefault("aws.region", "us-west-2"));
```

Comment:
I think we should be explicit and if this is not set have the user pass the region property via spark shell, etc